Add support for Jetson Orin NX

Austriker opened the pull request:
Enable support for a jetson Orin NX with NVMe

Based on this issue: https://hub.mender.io/t/jetson-orin-nx-and-jetpack-6-mender-intergration-issues/7375

- Create custom partition layout
- Use `<allocation_attribute> 0x808 </allocation_attribute>` to expand the user partition automatically during the flash process
- Add documentation

Austriker committed:
feat: add support for jetson orin nx

Signed-off-by: Hugo Serrat <hu.serrat@gmail.com>

fix: add documentation and fix kas config for orin nx

TheYoctoJester commented:
Hi @Austriker,

As you mentioned on IRC, I think changing to
```
PARTITION_FILE_EXTERNAL:p3768-0000-p3767-0000 = "${WORKDIR}/flash_l4t_t234_nvme_rootfs_ab.xml"
```
is a good approach. However it has the small shortcoming being tied to exactly this machine. So how about adding a short note to the layers README which explains why the line is there, and hints at adapting it for custom Orin NX based machines?

Austriker commented:
Hi @TheYoctoJester ,

I fixed the PR and added some documentation. I also remove the quick start guide since for me mender integration is not maintained anymore in tegrademo-distro.

TheYoctoJester commented:
Hi @Austriker,

I folded this PR into the bigger Tegra refactoring at https://github.com/mendersoftware/meta-mender-community/pull/439.
Can you check if this matches expectations, so we can eventually close here?
